src/Arachne.cc

```diff
@@ -667,11 +662,14 @@ dispatch() {
                 core.highestOccupiedContext++;
             } else {
                 currentIndex = 0;
-                // Reload the currentContext since we have reset currentIndex.
-                currentContext = core.localThreadContexts[currentIndex];
+                continue;
             }
         }
-        if (currentIndex == 0) {
+
+        // Ensure that we do not go out of bounds.
+        if (++currentIndex == maxThreadsPerCore) {
```

Comment:
If we check for `maxThreadsPerCore` instead of `0` in deciding whether to perform once-per-cycle operations (such as checking for requests from the Core Arbiter and accumulating idle time), we will not perform these operations until all thread slots are full.

Reply:
Yes correct. I will fix it.